### PR TITLE
chore: Add React PropTypes for more linting goodness.

### LINKED ISSUE #105: Add React propTypes?
Turns out that the [**eslint-plugin-react**](https://github.com/yannickcr/eslint-plugin-react#recommended) plugin includes a [set of recommended rules](https://github.com/yannickcr/eslint-plugin-react/blob/fcacb51579154e4acb3da416b805fad4bd1a4b94/index.js#L100-L125):

```json
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
```

```sh
$ npm run lint:js

> snoozetabs@1.0.0 lint:js /Users/pdehaan/dev/github/mozilla/SnoozeTabs
> eslint --max-warnings=0 .


/Users/pdehaan/dev/github/mozilla/SnoozeTabs/src/lib/components/DatePickerPanel.js
   13:27  error  'defaultValue' is missing in props validation  react/prop-types
   23:13  error  'id' is missing in props validation            react/prop-types
   23:17  error  'header' is missing in props validation        react/prop-types
   23:25  error  'active' is missing in props validation        react/prop-types
   23:33  error  'onClose' is missing in props validation       react/prop-types
  106:13  error  'onSelect' is missing in props validation      react/prop-types

/Users/pdehaan/dev/github/mozilla/SnoozeTabs/src/lib/components/MainPanel.js
  19:13  error  'id' is missing in props validation                  react/prop-types
  19:17  error  'active' is missing in props validation              react/prop-types
  19:25  error  'switchPanel' is missing in props validation         react/prop-types
  55:13  error  'scheduleSnoozedTab' is missing in props validation  react/prop-types
  69:13  error  'scheduleSnoozedTab' is missing in props validation  react/prop-types

/Users/pdehaan/dev/github/mozilla/SnoozeTabs/src/lib/components/ManagePanel.js
   20:13  error  'id' is missing in props validation                react/prop-types
   20:17  error  'entries' is missing in props validation           react/prop-types
   20:26  error  'active' is missing in props validation            react/prop-types
   20:34  error  'switchPanel' is missing in props validation       react/prop-types
   20:47  error  'tabIsSnoozable' is missing in props validation    react/prop-types
   96:13  error  'openSnoozedTab' is missing in props validation    react/prop-types
   96:29  error  'cancelSnoozedTab' is missing in props validation  react/prop-types
  121:18  error  'openSnoozedTab' is missing in props validation    react/prop-types
  141:13  error  'updateSnoozedTab' is missing in props validation  react/prop-types

/Users/pdehaan/dev/github/mozilla/SnoozeTabs/src/lib/components/SnoozePopup.js
  8:13  error  'activePanel' is missing in props validation  react/prop-types

✖ 21 problems (21 errors, 0 warnings)


npm ERR! Darwin 16.3.0
npm ERR! argv "/Users/pdehaan/.nvm/versions/node/v6.9.4/bin/node" "/Users/pdehaan/.nvm/versions/node/v6.9.4/bin/npm" "run" "lint:js"
npm ERR! node v6.9.4
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! snoozetabs@1.0.0 lint:js: `eslint --max-warnings=0 .`
npm ERR! Exit status 1
```

Looks like the only rule we're failing is [**react/prop-types**](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md). ([Docs](https://facebook.github.io/react/docs/typechecking-with-proptypes.html))

### PR DESCRIPTION
**Pull request by bwinton**
Fixes #105.

### CODE REVIEW

File: src/lib/components/SnoozePopup.js [hunk 1]
```diff
@@ -83,3 +83,8 @@ export default class SnoozePopup extends React.Component {
     }
   }
 }
+
+SnoozePopup.propTypes = {
```
Comment (lmorchard) [inline src/lib/components/SnoozePopup.js]: Got a bunch more functions we could add here, [as it's rendered from `snooze-popup.js`](https://github.com/bwinton/SnoozeTabs/blob/master/src/popup/snooze-content.js#L100)

Comment (bwinton) [inline src/lib/components/SnoozePopup.js]: We could, but they aren't used in this class, and they don't throw an error if they aren't listed, so I don't know if we should…

Comment (lmorchard) [inline src/lib/components/SnoozePopup.js]: They're not used here, but they *are* [passed](https://github.com/bwinton/SnoozeTabs/blob/470d69fe7f9bfe263dc4addd5d59da2819db6e91/src/lib/components/SnoozePopup.js#L65) & used & required by all the components this component uses.

Comment (bwinton) [inline src/lib/components/SnoozePopup.js]: Okay, fixed!

**Pull request merged**